lib/main.dart

```diff
@@ -36,72 +36,91 @@ class _RandomWordsState extends State<RandomWords> {
-        if (item.isOdd) return const Divider();
-        final index = item ~/ 2;
-        if (index >= _suggestions.length) {
-          _suggestions.addAll(generateWordPairs().take(10));
-        }
-        return _buildRow(_suggestions[index]);
+        return (item < _suggestions.length)
+            ? _buildRow(_suggestions[item])
+            : Container(color: Colors.white);
       },
     );
   }
 
   Widget _buildRow(WordPair pair) {
     final alreadySaved = _saved.contains(pair);
-    return ListTile(
-      title: Text(
-        pair.asPascalCase,
-        style: _biggerFont,
-      ),
-      trailing: Icon(
-        alreadySaved ? Icons.favorite : Icons.favorite_border,
-        color: alreadySaved ? Colors.red : null,
-      ),
-      onTap: () {
-        setState(() {
-          if (alreadySaved) {
-            _saved.remove(pair);
-          } else {
-            _saved.add(pair);
-          }
-        });
-      },
-    );
-  }
-  void _pushSaved() {
-    Navigator.of(context).push(
-      MaterialPageRoute<void>(
-        builder: (BuildContext context) {
-          final tiles = _saved.map(
-              (WordPair pair) {
-                return ListTile(
-                  title: Text(
-                    pair.asPascalCase,
-                    style: _biggerFont,
-                  ),
-                );
-              }
-          );
-          final divided = tiles.isNotEmpty
-              ? ListTile.divideTiles(context: context, tiles: tiles).toList()
-              : <Widget>[];
-          return Scaffold(
-            appBar: AppBar(
-              title: Text('Saved Suggestions'),
+    return (alreadySaved)
+        ? Container(color: Colors.white)
+        : ListTile(
+            title: Text(
+              pair.asPascalCase,
+              style: _biggerFont,
             ),
-            body: ListView(children: divided),
+            trailing: Icon(
+              alreadySaved ? Icons.favorite : Icons.favorite_border,
+              color: alreadySaved ? Colors.red : null,
+            ),
+            onTap: () {
+              setState(() {
+                if (alreadySaved) {
+                  _saved.remove(pair);
+                } else {
+                  _saved.add(pair);
+                }
+              });
+            },
           );
-        }
-      )
-    );
   }
-}
-
 
+  void _pushSaved() {
+    Navigator.of(context)
+        .push(MaterialPageRoute<void>(builder: (BuildContext context) {
+      final tiles = _saved.map((WordPair pair) {
+        return ListTile(
+          title: Text(
+            pair.asPascalCase,
+            style: _biggerFont,
+          ),
+          trailing: const Icon(Icons.delete),
+          onTap: () {
+            if (_saved.contains(pair)) {
+              _saved.remove(pair);
+              setState(() {}); // refresh page
+              Navigator.pop(context);
+            }
+          },
+        );
+      });
+      final divided = tiles.isNotEmpty
```

Comment:
Would be better to sepecify what is divided here. Maybe it's better to rename this variable to something like "dividedTiles" or "dividedItems"